docker/docker_client.go

```diff
@@ -248,31 +259,52 @@ func newDockerClient(sys *types.SystemContext, registry, reference string) (*doc
 	// Check if TLS verification shall be skipped (default=false) which can
 	// either be specified in the sysregistriesv2 configuration or via the
 	// SystemContext, whereas the SystemContext is prioritized.
-	skipVerify := false
+	var sysContextSkipVerify *bool
 	if sys != nil && sys.DockerInsecureSkipTLSVerify != types.OptionalBoolUndefined {
 		// Only use the SystemContext if the actual value is defined.
-		skipVerify = sys.DockerInsecureSkipTLSVerify == types.OptionalBoolTrue
-	} else {
-		reg, err := sysregistriesv2.FindRegistry(sys, reference)
-		if err != nil {
-			return nil, errors.Wrapf(err, "error loading registries")
+		skipVerify := sys.DockerInsecureSkipTLSVerify == types.OptionalBoolTrue
+		sysContextSkipVerify = &skipVerify
+	}
+
+	// Find the registry
+	reg, err := sysregistriesv2.FindRegistry(sys, ref)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error loading registries")
+	}
+
+	// Create a fallback registry from the provided parameter
+	if reg == nil {
+		// The only source for InsecureSkipVerify can be the SystemContext
+		insecure := false
+		if sysContextSkipVerify != nil && *sysContextSkipVerify {
+			insecure = true
 		}
-		if reg != nil {
-			skipVerify = reg.Insecure
+
+		// Retrieve the docker client from the found registry
+		reg = &sysregistriesv2.Registry{
+			Endpoint: sysregistriesv2.Endpoint{
+				URL:      registry,
+				Insecure: insecure,
+			},
 		}
 	}
-	tr.TLSClientConfig.InsecureSkipVerify = skipVerify
 
+	// Rewrite the registry if we found the dockerHostname
+	if registry == dockerHostname {
+		reg.URL = dockerRegistry
+	}
+
+	logrus.Debugf("Using registry from configuration: %+v", reg)
```

Comment:
The log doesn't match the `reg == nil` case. It could mislead into thinking that the registry from the fallback is configured in the `registries.conf`.